crates/cubecl-core/src/frontend/operation/base.rs

```diff
@@ -55,34 +43,21 @@ pub(crate) fn binary_expand_no_vec<F>(
 where
     F: Fn(BinaryOperator) -> Operator,
 {
-    let lhs_var: Variable = *lhs;
-    let rhs_var: Variable = *rhs;
+    let lhs = lhs.consume();
+    let rhs = rhs.consume();
 
     let item_lhs = lhs.item();
-    let item_rhs = rhs.item();
 
     let item = Item::new(item_lhs.elem);
 
-    // We can only reuse rhs.
-    let out = if lhs.can_mut() && item_lhs == item {
-        lhs
-    } else if rhs.can_mut() && item_rhs == item {
-        rhs
-    } else {
-        context.create_local(item)
-    };
-
-    let out_var = *out;
+    let output = context.create_local_binding(item);
```

Comment:
If I do the following:

```rust
let mut out = x + y;
```
Do we call this function? Would it be problematic since the out value may be reused?

Reply:
Creating a mutable local calls `Init::init` on the initializer, which assigns it to a new mutable local variable. So in `wgpu` this would be
```rust
let _10 = _8 + _9;
l_1_6 = _10;
```
or in SPIR-V
```
%10 = AddOp %8 %9
StoreOp %10 %out_var
```
See this code
```rust
pub fn init_expand<F>(context: &mut CubeContext, input: ExpandElement, func: F) -> ExpandElement
where
    F: Fn(UnaryOperator) -> Operator,
{
    if input.can_mut() {
        return input;
    }

    let input_var: Variable = *input;
    let item = input.item();

    let out = context.create_local_variable(item);
    let out_var = *out;

    let op = func(UnaryOperator {
        input: input_var,
        out: out_var,
    });

    context.register(op);

    out
}
```
`can_mut` will always return false for plain variables (i.e. local bindings).

Reply:
Perfect, that answer my question :)